python_modules/dagster/dagster/core/execution.py

```diff
@@ -534,6 +517,23 @@ def __init__(self, pipeline, errors, config_value, *args, **kwargs):
         super(PipelineConfigEvaluationError, self).__init__(error_msg, *args, **kwargs)
 
 
+def execute_plan(pipeline, execution_plan, environment=None, subset_info=None):
+    check.inst_param(pipeline, 'pipeline', PipelineDefinition)
+    check.inst_param(execution_plan, 'execution_plan', ExecutionPlan)
+    check.opt_dict_param(environment, 'environment')
+    check.opt_inst_param(subset_info, 'subset_info', ExecutionSubsetInfo)
+
+    typed_environment = create_typed_environment(pipeline, environment)
+
+    with yield_context(pipeline, typed_environment) as context:
+        true_execution_plan = create_subplan(
+            ExecutionPlanInfo(context=context, pipeline=pipeline, environment=typed_environment),
+            execution_plan,
+            subset_info,
+        ) if subset_info else execution_plan
+        return list(execute_plan_core(context, true_execution_plan))
```

Comment:
why list() and not for result in... yield?

Reply:
yeah i guess in all my test cases i was doing a list() construction. we can easily change if you feel strongly